llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -188,6 +188,13 @@ void ModuleShaderFlags::initialize(Module &M, DXILResourceTypeMap &DRTM,
         continue;
       }
 
+      // Set UseNativeLowPrecision using dx.nativelowprec module metadata
+      if (auto *NativeLowPrec = mdconst::extract_or_null<ConstantInt>(
+              M.getModuleFlag("dx.nativelowprec")))
+        if (MMDI.DXILVersion >= VersionTuple(1, 2) &&
```

Comment:
So I know the frontend likely has checks to confirm SM 6.2 or greater for this flag. If the backend is dxil-pc-shadermodel6.1 what would happen in a test case like `use-native-low-precision.ll`? Would we just expect to not see `!0 = !{i32 1, !"dx.nativelowprec", i32 1}`?

Reply:
The frontend is what is adding `!0 = !{i32 1, !"dx.nativelowprec", i32 1}`, so we wouldn't see it in the module.

If it is manually added to a module that is SM 6.1 or older, then it should have no effect on the shader flags.
Should I add a test for this?

Reply:
My concern was if the validator would allow the dxil if it was less than sm 6.2 and we saw `nativelowprec` set. The backend would be our last place to catch that and avoid the validator complaining.

If that isn't a concern then I don't think we need a test.

Reply:
I don't think the validator interprets any module metadata flag named "dx.nativelowpres", so it would just be ignored if it's left in the DXIL even if the SM is lower than 6.2.